registry/remote/repository_test.go

```diff
@@ -1207,6 +1207,71 @@ func Test_BlobStore_Fetch_Seek(t *testing.T) {
 	}
 }
 
+func Test_BlobStore_Fetch_ZeroSizedBlob(t *testing.T) {
+	blob := []byte("")
+	blobDesc := ocispec.Descriptor{
+		MediaType: "test",
+		Digest:    digest.FromBytes(blob),
+		Size:      int64(len(blob)),
+	}
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			t.Errorf("unexpected access: %s %s", r.Method, r.URL)
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		switch r.URL.Path {
+		case "/v2/test/blobs/" + blobDesc.Digest.String():
+			if rangeHeader := r.Header.Get("Range"); rangeHeader != "" {
+				if blobDesc.Size <= 0 {
+					t.Errorf("unexpected range header")
+					w.WriteHeader(http.StatusBadRequest)
+					return
+				}
+				rangeLimit, err := strconv.ParseInt((rangeHeader[8:]), 10, 64)
+				if err == nil && rangeLimit != blobDesc.Size-1 {
```

Comment:
What if `err != nil`? Should we return `400` in this case?

Reply:
Also, the `blobDesc.Size` is always `0` since `blob := []byte("")`. Thus, `rangeLimit != blobDesc.Size-1` translates to `rangeLimit != -1` and is always `true`.

Reply:
I kept `err == nil` for the (unlikely) case of "Range: bytes=0-", which is acceptable according to line 623: "The form of "Range: bytes=<start>-" is also acceptable."

I don't know if I need to further differentiate the cases where `err != nil`.

Reply:
When blobDesc.Size = 0, the first if statement returns. For whatever reaches the rangeLimit part, the size is >0.

Reply:
> When blobDesc.Size = 0, the first if statement returns. For whatever reaches the rangeLimit part, the size is >0.

Nice call out. As I've mentioned above `blobDesc.Size` is always `0`, thus `blobDesc.Size <= 0` is always true. Therefore, the lines 1232 - 1236 are never executed, and thus can be removed.

Reply:
Removed these lines.